usvm-ts/src/test/kotlin/org/usvm/util/TSTestResolver.kt

```diff
@@ -52,10 +59,48 @@ class TSTestResolver {
         }
     }
 
-    private fun resolveExpr(expr: UExpr<out USort>, type: EtsType): TSObject = when (type) {
-        is EtsPrimitiveType -> resolvePrimitive(expr, type)
-        is EtsRefType -> TODO()
-        else -> TODO()
+    private fun resolveParams(
+        params: List<EtsMethodParameter>,
+        ctx: TSContext,
+        model: UModelBase<EtsType>,
+    ): List<TSObject> = with(ctx) {
+        params.mapIndexed { idx, param ->
+            val type = param.type
+            val lValue = URegisterStackLValue(typeToSort(type), idx)
+            val expr = model.read(lValue).extractOrThis()
+            if (type is EtsUnknownType) {
+                approximateParam(expr.cast(), idx, model)
+            } else resolveExpr(expr, type, model)
+        }
+    }
+
+    private fun approximateParam(expr: UConcreteHeapRef, idx: Int, model: UModelBase<EtsType>): TSObject =
+        with(expr.ctx as TSContext) {
+            val suggestedType = state.getSuggestedType(idx)
+            return suggestedType?.let { newType ->
```

Comment:
Something odd is happening here. You can always access all required information using symbolic values and symbolic values only, otherwise you will encounter path divergence. This map **cannot** be built from concrete values to type info, only from symbolic ones

Reply:
I believe different types of ref entities (reg readings, etc.) have unique ways to generate a key. For example, reg readings can use its' ```idx``` field as a key, and this key leads to correct list of types from any model.

Reply:
But you might store it with another key -- if I resolve something from registerReading(1), it doesn't mean that I always accessed this value using this key. Probably, I could do it with registerReading(2).getByIndex(3), i could have accessed it with RR(3) (because of aliasing) and many many other ways